charmcraft.yaml

```diff
@@ -4,14 +4,22 @@
 type: charm
 parts:
   charm:
+    build-snaps:
+      - rustup
     charm-binary-python-packages:
       - setuptools
     build-packages:
       - pkg-config
       - libffi-dev
       - libssl-dev
-      - rustc
-      - cargo
+    override-build: |
+      rustup default stable
```

Comment:
With this, we can drop the constraint on rdps-py in pyproject.toml, courtesy of the sql team :100: